### ISSUE TITLE
Will TL support accepting placeholder for `is_train` in batch normalization?

### ISSUE THREAD

**Issue by CTTC** [zsdonghao/tensorlayer#116]
As discussed in https://github.com/tensorflow/tensorflow/issues/1122, tf.slim supports passing tf.placeholder to `is_train`, will TL consider this feature as well?

**Comment by CTTC** [comment]
I try to implement something like:
```python
            with tf.variable_scope('bn'):
                is_train_bn = tl.layers.TimeDistributedLayer(net,
                                                             layer_class=tl.layers.BatchNormLayer,
                                                             args={'act': tf.nn.elu,
                                                                   'is_train': True,
                                                                   'name': 'fc1_bn_'},
                                                             name='time_dense_fc1_bn')
                tf.get_variable_scope().reuse_variables()
                tl.layers.set_name_reuse(True)
                is_test_bn = tl.layers.TimeDistributedLayer(net,
                                                            layer_class=tl.layers.BatchNormLayer,
                                                            args={'act': tf.nn.elu,
                                                                  'is_train': False,
                                                                  'name': 'fc1_bn_'},
                                                            name='time_dense_fc1_bn')
                net = tf.cond(self.is_training, is_train_bn, is_test_bn)
```

**Comment by zsdonghao** [comment]
we don't have plan to support this feature, as we suggest to build different inferenec for testing and training, [click](https://github.com/wagamamaz/tensorlayer-tricks)
